### PR TITLE
feat: Redesign of the main docs landing page and typography unification across the docs

### PR DESCRIPTION
**Pull request by jbartadev**
Snímek obrazovky 2023-08-17 v 11 21 18

Snímek obrazovky 2023-08-17 v 11 22 04

### CODE REVIEW

**Comment by barjin** [comment]
Oof, just copied the `custom.css` stylesheet to one of the side projects (Apify SDK JS). It seems that the biggest problem is heading size that's specified in `rems`, though - and in Docusaurus, heading sizes are stored in the `--ifm-h1-font-size` CSS variables. It's the same story with Docusaurus-defined paddings / margins.

Btw you get the same problem in the main Docusaurus instance (`apify-docs`) when you go to one of the docs pages (Academy / Platform) - that way, you get HMR while editing the css at least 😬

https://github.com/apify/apify-docs/assets/61918049/25410f39-0723-42b6-acae-9282d37846fc

**Comment by jbartadev** [comment]
> Oof, just copied the `custom.css` stylesheet to one of the side projects (Apify SDK JS). It seems that the biggest problem is heading size that's specified in `rems`, though - and in Docusaurus, heading sizes are stored in the `--ifm-h1-font-size` CSS variables. It's the same story with Docusaurus-defined paddings / margins.
> 
> Btw you get the same problem in the main Docusaurus instance (`apify-docs`) when you go to one of the docs pages (Academy / Platform) - that way, you get HMR while editing the css at least 😬
> 
>  Peek.2023-08-16.14-34.mp4

@barjin Do you know if is there a way to change the values of the Docusaurus CSS variables?

**Comment by barjin** [comment]
I think it should be as simple as redefining them in the theme's `custom.css` file (we're already doing it, [see here](https://github.com/apify/apify-docs/blob/e4e26f57f4ff30526619224f443f99e001fa035e/apify-docs-theme/src/theme/custom.css#L38)) - haven't tried it though.

File: .github/workflows/publish-theme.yaml
Comment (B4nan) [inline .github/workflows/publish-theme.yaml:48]: I would guess this needs to stay, we need to be able to push to git from here. Same with the NPM thingy, that is so we can publish new themes to the `@apify/` namespace.

File: apify-docs-theme/src/theme/custom.css
Comment (B4nan) [inline apify-docs-theme/src/theme/custom.css:607]: This looks wrong, the plugin-docs class is in the html element unless you changed something.

File: .github/workflows/publish-theme.yaml
Comment (jbartadev) [reply .github/workflows/publish-theme.yaml]: Ha, you're right. My bad, I thought it was connected only to the installation.

**Review by drobnikj — APPROVED** [review]
I like the new design!

File: apify-docs-theme/src/theme/custom.css
Comment (jbartadev) [reply apify-docs-theme/src/theme/custom.css:607]: I thought so too but then I noticed it without the spaced it overrides all H1s everywhere and with the space it affects only the `.plugin-docs class`

**Comment by B4nan** [comment]
Looks like this got somehow forgotten. I guess it was waiting for us to merge (while we are generally used to let author merge their own PRs once they are approved).

Would be good if someone rebases it.

**Comment by barjin** [comment]
Unfortunately, this PR still breaks the styling pretty much everywhere in the docs. The only culprit is the `font-size: 10px` thing on the root `html` tag... I guess we cannot get rid of it soon @jbartadev , right? :/ 

> Flipping through the current docs and this PR:

Peek 2023-10-04 13-58

> Closer look at a random page with this PR changes:

obrazek

To me, it's mostly the smaller paddings + font sizes all over the page.

After merging the latest master, this PR also messes up the search modal :'(

File: apify-docs-theme/src/theme/custom.css
Comment (B4nan) [inline apify-docs-theme/src/theme/custom.css:168]: FYI this one is missing a value

Comment (jbartadev) [reply apify-docs-theme/src/theme/custom.css:168]: My bad, this is still in progress so you can just ignore the changes until we request the review again.

CC @webrdaniel

File: apify-docs-theme/src/theme/custom.css
Comment (OPesicka) [inline apify-docs-theme/src/theme/custom.css:170]: These don't get applied in the docs. Even when I tried adding `!imporant` the font-size is still overridden by `.markdow`

image

File: apify-docs-theme/src/theme/custom.css
Comment (OPesicka) [inline apify-docs-theme/src/theme/custom.css:127]: We should get rid of the `Lota Grotesque` font family in favor of system fonts.

File: apify-docs-theme/src/theme/custom.css
Comment (OPesicka) [inline apify-docs-theme/src/theme/custom.css:168]: The font sizes are correct but when we were defining them we forgot that H1 basically doesn't exist in markdown and you use H2 as the "main" heading. So we should push down the font-sizes so that H1 size is not H2 and come up with a bigger H1 size.

File: src/pages/index.module.css
Comment (OPesicka) [inline src/pages/index.module.css:29]: Just a nit the titles in these cards are slightly misaligned 
image

**Review by OPesicka — COMMENTED** [review]
Did a first pass, but found some issues that need fixing before I dig deeper (mainly the heading sizes)

File: apify-docs-theme/src/theme/custom.css
Comment (OPesicka) [inline apify-docs-theme/src/theme/custom.css:897]: This doesn't seem right. If ti only affects these cards (viz screenshot) I'd go for 16px
image

File: apify-docs-theme/src/theme/custom.css
Comment (OPesicka) [inline apify-docs-theme/src/theme/custom.css:597]: The padding is too small now. I'd do 12/16px 
image

**Review by OPesicka — APPROVED** [review]
2 nits then it's ready to go from my side 🚀

**Pull request merged**
